cep-sigstore-predicate.md

```diff
@@ -12,61 +12,250 @@
+
+Attestations that present this metadata in a cryptographically
+verifiable manner are desirable for a number of reasons:
+
+* Package maintainers wish to demonstrate the integrity and authenticity
+  of their package uploads;
+* Individual downstream users wish to verify the integrity and authenticity of
+  packages they consume, without placing additional trust in the
+  channel or channel's hosting server;
+* Attestations change the sophistication and risk profile for attackers in
+  defenders' favor: the attacker must be sufficiently sophisticated
+  to access private key material, *and* have a risk tolerance profile that
+  accepts exposure via auditable transparency logs.
+
+More broadly, attestation schemes like the one proposed in this CEP have
+seen adoption in similar and related ecosystems:
+
+* Python (PyPA/PyPI): [PEP 740] and [PyPI - Attestations]
+* NodeJS (npm): [npm - Generating provenance statements]
+* Ruby (RubyGems): [rubygems/release-gem]
 
 ## Specification
 
-The in-toto predicate should contain the following fields:
+### Attestation format
+
+This CEP proposes the following attestation statement layout, using the
+[in-toto Statement schema]:
+
+- `predicateType` **MUST** be `https://schemas.conda.org/attestations/publish/v1`
+- `subject` **MUST** be a single [`ResourceDescriptor`], with the following
+  constraints:
+    - `subject[0].name` **MUST** be the full filename of the conda package
+      that will be part of the `repodata.json` and under which it will appear on
+      the server.
+    - `subject[0].digest` **MUST** be a [`DigestSet`], and it **MUST** contain
+      a single `sha256` entry with the SHA256 hash of the conda package.
+- `predicate` **MAY** be present. If present and not `null`, it **MUST** be a
+  JSON object with the following fields:
+    - `targetChannel` **MUST** be a string, indicating where the package
+      is being uploaded to. This field **MUST** be a valid URL with no
+      trailing slashes.
+
+An example of a compliant statement is provided below:
 
 ```json
 {
     "_type": "https://in-toto.io/Statement/v0.1",
     "subject": [{
         "name": "file-name-0.0.1-h123456_5.conda",
-        "digest": {"sha256": "..."}, ...
+        "digest": {"sha256": "01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546b"},
     }],
-    // Schema URL
-    "predicateType": "https://schemas.conda.org/predicate-v1.json",
+    "predicateType": "https://schemas.conda.org/attestations/publish/v1",
     "predicate": {
-        // Canonical URL of the target channel
         "targetChannel": "https://prefix.dev/conda-forge",
     }
 }
 ```
 
-The `subject` field is already defined in the in-toto specification and contains the name of the package and its digest.
-For conda packages a SHA256 hash MUST be used.
-The subject MUST be the full filename of the conda package that will be part of the repodata.json and under which it will appear on the server.
+### Signing and distributing
```

Comment:
@wolfv PTAL at this section and the `Verifying` section below it! I've tried to keep things detailed but at a pretty high level, i.e. not going into the details of Sigstore's internal signing/verifying steps or cryptographic selections.